api/schemas/report.json

```diff
@@ -114,6 +114,11 @@
             ]
         },
 
+        "master": {
+            "description": "The name of the master that was used to compile the catalog. If failover occurred, this will hold the first master successfully contacted. If this run had no master (e.g. a `puppet apply` run), this field will be blank",
```

Comment:
Ah right, that probably didn't work because you didn't specify a separate (functional) report server. Normally, if no masters could be contacted, that would mean that the default report server also couldn't be contacted (it's $server), and hence no report.

The report is actually always first generated agent-side, so we're guaranteed to have one during the run. If you set `report_server` to a working server (that isn't in your master list), you should be able to get it. It will probably have a lot of empty fields.

In the case where the user set `server` and not `server_list`, we'd have to check the `masterport` setting for the port. It will have its default value if it wasn't set.